windows/win_get_url.ps1

```diff
@@ -67,21 +67,31 @@ If ($force -or -not (Test-Path $dest)) {
 }
 Else {
     Try {
-        $webRequest = [System.Net.HttpWebRequest]::Create($url)
-
+        #before download, head to compare -> more efficient
+        $webRequestHead = [System.Net.HttpWebRequest]::Create($url)
         if($username -and $password){
-            $webRequest.Credentials = New-Object System.Net.NetworkCredential($username, $password)
+                $webRequestHead.Credentials = New-Object System.Net.NetworkCredential($username, $password)
         }
+        $webRequestHead.Method = "HEAD"
+        [System.Net.HttpWebResponse]$webResponseHead = $webRequestHead.GetResponse()
+        $modified = $webResponseHead.LastModified
+        $webRequestHead = $null
 
-        $webRequest.IfModifiedSince = ([System.IO.FileInfo]$dest).LastWriteTime
-        $webRequest.Method = "GET"
-        [System.Net.HttpWebResponse]$webResponse = $webRequest.GetResponse()
-        
-        $stream = New-Object System.IO.StreamReader($response.GetResponseStream())
-        
-        $stream.ReadToEnd() | Set-Content -Path $dest -Force -ErrorAction Stop
-        
-        $result.changed = $true
+        #comparision
+        if ($webResponseHead.LastModified -ne ([System.IO.FileInfo]$dest).LastWriteTime) {
+
+                $webRequest = [System.Net.HttpWebRequest]::Create($url)
```

Comment:
Looking again at my notes on https://github.com/ansible/ansible-modules-core/issues/2146 I had problems with using this method to download binary (zip) files.  So I think it would be better to use System.Net.WebClient which has a DownloadFile method ( see comment on 2146 for usage).